react/features/invite/components/InviteButton.web.js

```diff
@@ -0,0 +1,231 @@
+
+        return (
+            <div className = 'filmstrip__invite'>
+                <div className = 'invite-button-group'>
+                    <Button
+                        onClick = { this._onInviteClick }
+                        shouldFitContainer = { true }>
+                        { t('invite.invitePeople') }
+                    </Button>
+                    { this.props._isDialOutAvailable
+                        || this.props._isAddToCallAvailable
+                        ? <DropdownMenu
+                            items = { this.state.inviteOptions }
+                            onItemActivated = { this._onInviteOptionSelected }
+                            position = { VERTICAL_FILMSTRIP
+                                ? 'bottom right'
+                                : 'top right' }
+                            shouldFlip = { true }
+                            triggerType = 'button' />
+                        : null }
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Handles the click of the invite button.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onInviteClick() {
+        this.props.openDialog(InviteDialog);
+    }
+
+    /**
+     * Handles selection of the invite options.
+     *
+     * @param { Object } option - The invite option that has been selected from
+     * the dropdown menu.
+     * @private
+     * @returns {void}
+     */
+    _onInviteOptionSelected(option) {
+        this.state.inviteOptions[0].items.forEach(item => {
+            if (item.content === option.item.content) {
+                item.action();
+            }
+        });
+    }
+
+    /**
+     * Updates the invite items list depending on the availability of the
+     * features.
+     *
+     * @param {Object} props - The read-only properties of the component.
+     * @private
+     * @returns {void}
+     */
+    _updateInviteItems(props) {
+        const { t } = this.props;
+
+        const inviteItems = [];
+
+        inviteItems.splice(
+            getInviteOptionPosition(SHARE_LINK_OPTION),
+            0,
+            {
+                content: t('toolbar.invite'),
+                action: () => this.props.openDialog(InviteDialog)
+            }
+        );
+
+        if (props._isDialOutAvailable) {
+            inviteItems.splice(
+                getInviteOptionPosition(DIAL_OUT_OPTION),
+                0,
+                {
+                    content: t('dialOut.dialOut'),
+                    action: () => this.props.openDialog(DialOutDialog)
+                }
+            );
+        }
+
+        if (props._isAddToCallAvailable) {
+            inviteItems.splice(
+                getInviteOptionPosition(ADD_TO_CALL_OPTION),
+                0,
+                {
+                    content: interfaceConfig.ADD_PEOPLE_APP_NAME,
+                    action: () => this.props.openDialog(AddPeopleDialog)
+                }
+            );
+        }
```

Comment:
Why prefer splice over doing a push? Check for featureAvailable, push to array.

Reply:
Because I want to insert the element at the specific position defined in interfaceConfig.

Reply:
Ah okay, I'm seeing it now. I thought you were just always splicing to the front. It makes sense then. An alternative might be to map or filter over the array and maybe keep a map somewhere of expected invite option and its respective enable check as well as the dropdown option.

Reply:
Is splice that bad? :)

Reply:
No it's not bad.